Recipes/Recipes/View Controllers/RecipesTableViewController.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  RecipesTableViewController.swift
+//  Recipes
+//
+//  Created by morse on 5/6/19.
+//  Copyright © 2019 Lambda Inc. All rights reserved.
+//
+
+import UIKit
+
+class RecipesTableViewController: UITableViewController {
+    var recipes: [Recipe] = [] {
+        didSet {
+            print("Recipes set: \(recipes.count)")
+            tableView.reloadData()
+        }
+    }
+//    var recipeIndexPath: Int?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+    }
+
+    // MARK: - Table view data source
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        print("#OfRows: \(recipes.count)")
+        return recipes.count
+    }
+
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "RecipeTitleCell", for: indexPath)
+        
+        cell.textLabel?.text = recipes[indexPath.row].name
+        
+        return cell
+    }
+ 
+    // MARK: - Navigation
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "DetailSegue" {
+            
```

Comment:
Your implementation works just fine, but here I would replace lines 46-48 
```
let recipesDetailVC = segue.destination as! RecipeDetailViewController
recipesDetailVC.recipe = recipes[tableView.indexPathForSelectedRow!.row]
```
mainly because if either of these checks failed, I want the app to crash here. in your implementation, you could accomplish this by throwing a descriptive fatalError() like so 

```
guard let recipesDetailVC = segue.destination as? RecipeDetailViewController else { fatalError("destination VC is not of type RecipeDetailViewController") }

guard let recipeIndexPath = tableView.indexPathForSelectedRow else { fatalError("invalid such and such with some cryptic error code") }
```